enterprise/tailnet/pgcoord_test.go

```diff
@@ -0,0 +1,656 @@
+package tailnet_test
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/dbtestutil"
+	"github.com/coder/coder/enterprise/tailnet"
+	agpl "github.com/coder/coder/tailnet"
+	"github.com/coder/coder/testutil"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+func TestPGCoordinatorSingle_ClientWithoutAgent(t *testing.T) {
```

Comment:
Maybe it's just personal preference, but is there a reason you don't do nested tests?

Reply:
Nesting increases the indent, and therefore leaves less horizontal room for readability, and https://github.com/coder/coder/pull/5376